web/modules/custom/collection/src/ContextProvider/CollectionRouteContext.php

```diff
@@ -48,6 +51,22 @@ public function getRuntimeContexts(array $unqualified_context_ids) {
         }
       }
     }
+    elseif ($route_parameters = $this->routeMatch->getParameters()) {
+      /** @var ContentEntityInterface $route_parameter */
+      foreach ($route_parameters as $route_parameter) {
```

Comment:
Hmm ok this is apparently how the `ParameterBag` works, the upcasted entities on the route are accessed by their parameter name. This is different from Drupal 7, where you could specify the 'position' of the object. If you did a call to `menu_get_object()` in D7 you would only get an object back if it adhered to the route `entity_type/%id`, since it would ignore any entity IDs that were not in position "1". Now it matches on any entity found, whatever its position in the string.

If we are not sure of the position of the object on the route then unexpected things might happen, especially in the backend. For example we might be on some administration page that happens to have a collection content entity in it, for example some imaginary route like 'admin/config/subscriptions/collection/%id' - this would match, making the collection available on the route, and unexpected things will happen (for example the "Join collection" button might be shown).

So if we cannot rely on the RouteMatch giving us back only the "main entity" we need to come up with some other way to limit this to the correct routes. I'm guessing we should provide a whitelist of routes, let's start with `node/{node}`, `rdf_entity/{rdf_entity}` and `rdf_entity/{rdf_entity}/leave`. In the future we will need to add routes for the overviews and search results pages too.